src/pages/SignUp.tsx

```diff
@@ -75,9 +75,31 @@ const defaultData = {
 };
 
 const SignUp = () => {
+  const [steps, setSteps] = useState(defaultSteps);
   const [formData, setForm] = useForm(defaultData); // useForm is an extension of React hooks to manage form state
-  const { step, navigation } = useStep({ steps: flatSteps as any }); // defaults for our steps
-  const [landingStep, setLandingStep] = useState(1); // Landing has two steps internally, based on if "individual"
+  const { step, navigation } = useStep({
+    steps: flatSteps(steps) as any
+  }); // defaults for our defaultSteps
+  const [landingStep, setLandingStep] = useState(1); // Landing has two defaultSteps internally, based on if "individual"
+
+  useEffect(() => {
+    const newSteps = [...steps];
+    const stageRole = formData.stageRole;
+
+    if (stageRole === '') {
+      return;
+    }
+
+    if (stageRole === 'on-stage') {
+      newSteps.splice(5, 1);
+    } else if (stageRole === 'off-stage') {
+      newSteps.splice(4, 1);
+    }
+
+    setSteps(newSteps);
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [formData.stageRole]);
```

Comment:
You don't need the full wrapping comments for eslint-disable, you can just add one line like this:
```
}, [formData.stageRole]); // eslint-disable-line react-hooks/exhaustive-deps
```

Reply:
Ah, good to know. Learned something new for the future :)